pandas/tests/frame/methods/test_to_csv.py

```diff
@@ -1164,7 +1177,11 @@ def test_to_csv_with_dst_transitions(self, td, temp_file):
         # we have to reconvert the index as we
         # don't parse the tz's
         result = read_csv(path, index_col=0)
-        result.index = to_datetime(result.index, utc=True).tz_convert("Europe/London")
+        result.index = (
+            to_datetime(result.index, utc=True)
+            .tz_convert("Europe/London")
+            .as_unit("ns")
```

Comment:
What part of this test makes it so nanosecond inference is the right approach?

Reply:
Yeah I'm curious about this too. Is `df.index.unit` not `"s"`?

Reply:
we do get "s" at the read_csv step.  Converting back to "ns" is bc in the next line we compare with the original `df` which has nano.

Reply:
Ah I didn't realize `times` had "ns" unit (I had assumed it was "s")